crates/rpc/engine/fork_choice.rs

```diff
@@ -0,0 +1,55 @@
+use ethereum_rust_storage::Store;
+use serde_json::{json, Value};
+
+use crate::{
+    types::fork_choice::{ForkChoiceState, PayloadAttributesV3},
+    RpcErr,
+};
+
+#[derive(Debug)]
+pub struct ForkChoiceUpdatedV3 {
+    pub fork_choice_state: ForkChoiceState,
+    #[allow(unused)]
+    pub payload_attributes: Option<PayloadAttributesV3>,
+}
+
+impl ForkChoiceUpdatedV3 {
+    pub fn parse(params: &Option<Vec<Value>>) -> Option<ForkChoiceUpdatedV3> {
+        let params = params.as_ref()?;
+        if params.len() != 2 {
+            return None;
+        }
+        Some(ForkChoiceUpdatedV3 {
+            fork_choice_state: serde_json::from_value(params[0].clone()).ok()?,
+            payload_attributes: serde_json::from_value(params[1].clone()).ok()?,
+        })
+    }
+}
+
+pub fn forkchoice_updated_v3(
+    request: ForkChoiceUpdatedV3,
+    storage: Store,
+) -> Result<Value, RpcErr> {
+    // Just a minimal implementation to pass rpc-compat Hive tests.
+    // TODO (#50): Implement `engine_forkchoiceUpdatedV3`
+    let safe = storage.get_block_number(request.fork_choice_state.safe_block_hash);
+    let finalized = storage.get_block_number(request.fork_choice_state.finalized_block_hash);
+
+    // Check if we already have the blocks stored.
```

Comment:
no need for this check, you might get multiple updates that have the same safe/finalized block. You don't need to return an error here.

Reply:
Hmmm, in this step I'm actually checking that we have those blocks stored beforehand, rather than if we have already marked them as safe/finalized. But maybe I didn't fully understand what you meant.

Reply:
Ahh, my bad then.